runtime/CSharp/src/Tree/Pattern/ParseTreePatternMatcher.cs

```diff
@@ -648,32 +647,34 @@ internal virtual IList<Chunk> Split(string pattern)
             // collect into chunks now
             if (ntags == 0)
             {
-                string text = Sharpen.Runtime.Substring(pattern, 0, n);
-                chunks.Add(new TextChunk(text));
+                chunks.Add(new TextChunk(pattern));
             }
             if (ntags > 0 && starts[0] > 0)
             {
                 // copy text up to first tag into chunks
-                string text = Sharpen.Runtime.Substring(pattern, 0, starts[0]);
+                string text = pattern.Substring(0, starts[0]);
                 chunks.Add(new TextChunk(text));
             }
             for (int i_1 = 0; i_1 < ntags; i_1++)
             {
                 // copy inside of <tag>
-                string tag = Sharpen.Runtime.Substring(pattern, starts[i_1] + start.Length, stops[i_1]);
+                int offset = starts[i_1] + start.Length;
```

Comment:
so now we have more code ? What's the benefit ?

Reply:
Getting rid of confusing. When .NET developer reads code, he expects some familiar behavior of methods with similar signatures (Java `substring` and C# `Substring` have the same naming but different semantics). Also, `Sharpen.Runtime.Substring` is used only in several places.

Reply:
I want to see the same algorithms in Java and C#, and that was the original thinking behind Sharpen I believe. That is way more important. I don't disagree with your observation, but then we should rename Substring to SubstringUsingOffset.

Reply:
> I don't disagree with your observation, but then we should rename Substring to SubstringUsingOffset.

I don't clearly understand. You mean reverting my changes and using renamed `Sharpen.Runtime.Substring` -> `Sharpen.Runtime.SubstringUsingOffset`? System `Substring` can no be renamed.

Reply:
I would rename Sharpen.Runtime.Substring and convert it to an extension method